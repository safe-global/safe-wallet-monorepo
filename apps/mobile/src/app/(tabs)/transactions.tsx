import { SafeTab } from '@/src/components/SafeTab'
import React from 'react'
import { TxHistoryContainer } from '@/src/features/TxHistory'
import { ComingSoon } from '@/src/components/ComingSoon/ComingSoon'
import { useNavigation } from 'expo-router'
import TransactionHeader from '@/src/features/TxHistory/components/TransactionHeader'

const tabItems = [
  {
    label: 'History',
    title: 'Transactions',
    Component: TxHistoryContainer,
  },
  {
    label: `Messages`,
    title: 'Messages',
    Component: ComingSoon,
  },
]

export default function TransactionScreen() {
  const navigation = useNavigation()

  const onIndexChange = (index: number) => {
    navigation.setOptions({
      headerTitle: () => <TransactionHeader title={tabItems[index].title} />,
    })
  }
  return <SafeTab items={tabItems} containerStyle={{ marginTop: 16 }} onIndexChange={onIndexChange} />
}
