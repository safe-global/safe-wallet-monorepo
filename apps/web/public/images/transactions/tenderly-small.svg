<svg width="17" height="16" viewBox="0 0 17 16" fill="none" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
<rect y="0.393555" width="17" height="15.2128" fill="url(#pattern0_6059_15659)"/>
<defs>
<pattern id="pattern0_6059_15659" patternContentUnits="objectBoundingBox" width="1" height="1">
<use xlink:href="#image0_6059_15659" transform="matrix(0.0248575 0 0 0.0277778 -0.0271173 0)"/>
</pattern>
<image id="image0_6059_15659" width="156" height="36" preserveAspectRatio="none" xlink:href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAJwAAAAkCAYAAAB8IVNVAAAMPmlDQ1BJQ0MgUHJvZmlsZQAASImVVwdYU8kWnluSkEBoAQSkhN4EkRpASggt9N5shCRAKDEGgoodXVRw7SICNnRVRMFKsyN2FsXeFwsqyrpYsCtvUkDXfeV7831z57//nPnPmXNn7r0DgNpxjkiUi6oDkCcsEMcG+9OTU1LppKeAAHBAASZAmcPNFzGjo8MBLEPt38u76wCRtlfspVr/7P+vRYPHz+cCgERDnM7L5+ZBfAAAvJorEhcAQJTyZlMLRFIMK9ASwwAhXiTFmXJcLcXpcrxHZhMfy4K4HQAlFQ5HnAmA6iXI0wu5mVBDtR9iRyFPIARAjQ6xT17eZB7EaRBbQxsRxFJ9RvoPOpl/00wf1uRwMoexfC6yohQgyBflcqb/n+n43yUvVzLkwxJWlSxxSKx0zjBvN3Mmh0mxCsR9wvTIKIg1If4g4MnsIUYpWZKQBLk9asDNZ8GcAR2IHXmcgDCIDSAOEuZGhiv49AxBEBtiuELQaYICdjzEuhAv4ucHxilsNoknxyp8oY0ZYhZTwZ/liGV+pb7uS3ISmAr911l8tkIfUy3Kik+CmAKxeaEgMRJiVYgd8nPiwhQ2Y4uyWJFDNmJJrDR+c4hj+cJgf7k+VpghDopV2Jfm5Q/NF9uUJWBHKvC+gqz4EHl+sHYuRxY/nAt2iS9kJgzp8POTw4fmwuMHBMrnjj3jCxPiFDofRAX+sfKxOEWUG62wx035ucFS3hRil/zCOMVYPLEALki5Pp4hKoiOl8eJF2VzQqPl8eDLQThggQBABxJY08FkkA0EnX1NffBO3hMEOEAMMgEf2CuYoRFJsh4hvMaBIvAnRHyQPzzOX9bLB4WQ/zrMyq/2IEPWWygbkQOeQJwHwkAuvJfIRgmHvSWCx5AR/MM7B1YujDcXVmn/v+eH2O8MEzLhCkYy5JGuNmRJDCQGEEOIQUQbXB/3wb3wcHj1g9UJZ+AeQ/P4bk94QugiPCRcI3QTbk0SFIt/ijICdEP9IEUu0n/MBW4JNV1xf9wbqkNlXAfXB/a4C/TDxH2hZ1fIshRxS7NC/0n7bzP44Wko7MiOZJQ8guxHtv55pKqtquuwijTXP+ZHHmv6cL5Zwz0/+2f9kH0ebMN+tsQWYfuxM9gJ7Bx2GGsCdOwY1ox1YEekeHh1PZatriFvsbJ4cqCO4B/+hp6sNJP5jnWOvY5f5H0F/GnSdzRgTRZNFwsyswroTPhF4NPZQq7DKLqTo5MzANLvi/z19SZG9t1AdDq+c/P/AMD72ODg4KHvXOgxAPa6w+3f8p2zZsBPhzIAZ1u4EnGhnMOlFwJ8S6jBnaYHjIAZsIbzcQJuwAv4gUAQCqJAPEgBE2H0WXCdi8FUMBPMAyWgDCwHa0Al2Ai2gB1gN9gHmsBhcAKcBhfAJXAN3IGrpwe8AP3gHfiMIAgJoSI0RA8xRiwQO8QJYSA+SCASjsQiKUgakokIEQkyE5mPlCErkUpkM1KL7EVakBPIOaQLuYU8QHqR18gnFENVUC3UELVER6MMlImGofHoBDQTnYIWoQvQpWgFWoPuQhvRE+gF9Brajb5ABzCAKWM6mAlmjzEwFhaFpWIZmBibjZVi5VgNVo+1wud8BevG+rCPOBGn4XTcHq7gEDwB5+JT8Nn4ErwS34E34u34FfwB3o9/I1AJBgQ7gieBTUgmZBKmEkoI5YRthIOEU3Av9RDeEYlEHaIV0R3uxRRiNnEGcQlxPbGBeJzYRXxEHCCRSHokO5I3KYrEIRWQSkjrSLtIx0iXST2kD0rKSsZKTkpBSqlKQqVipXKlnUpHlS4rPVX6TFYnW5A9yVFkHnk6eRl5K7mVfJHcQ/5M0aBYUbwp8ZRsyjxKBaWecopyl/JGWVnZVNlDOUZZoDxXuUJ5j/JZ5QfKH1U0VWxVWCrjVSQqS1W2qxxXuaXyhkqlWlL9qKnUAupSai31JPU+9YMqTdVBla3KU52jWqXaqHpZ9aUaWc1Cjak2Ua1IrVxtv9pFtT51srqlOkudoz5bvUq9Rf2G+oAGTWOMRpRGnsYSjZ0a5zSeaZI0LTUDNXmaCzS3aJ7UfETDaGY0Fo1Lm0/bSjtF69EiallpsbWytcq0dmt1avVra2q7aCdqT9Ou0j6i3a2D6VjqsHVydZbp7NO5rvNphOEI5gj+iMUj6kdcHvFed6Suny5ft1S3Qfea7ic9ul6gXo7eCr0mvXv6uL6tfoz+VP0N+qf0+0ZqjfQayR1ZOnLfyNsGqIGtQazBDIMtBh0GA4ZGhsGGIsN1hicN+4x0jPyMso1WGx016jWmGfsYC4xXGx8zfk7XpjPpufQKeju938TAJMREYrLZpNPks6mVaYJpsWmD6T0zihnDLMNstVmbWb+5sXmE+UzzOvPbFmQLhkWWxVqLMxbvLa0skywXWjZZPrPStWJbFVnVWd21plr7Wk+xrrG+akO0Ydjk2Ky3uWSL2rraZtlW2V60Q+3c7AR26+26RhFGeYwSjqoZdcNexZ5pX2hfZ//AQcch3KHYocnh5Wjz0amjV4w+M/qbo6tjruNWxztjNMeEjike0zrmtZOtE9epyumqM9U5yHmOc7PzKxc7F77LBpebrjTXCNeFrm2uX93c3cRu9W697ubuae7V7jcYWoxoxhLGWQ+Ch7/HHI/DHh893TwLPPd5/uVl75XjtdPr2VirsfyxW8c+8jb15nhv9u72ofuk+Wzy6fY18eX41vg+9DPz4/lt83vKtGFmM3cxX/o7+ov9D/q/Z3myZrGOB2ABwQGlAZ2BmoEJgZWB94NMgzKD6oL6g12DZwQfDyGEhIWsCLnBNmRz2bXs/lD30Fmh7WEqYXFhlWEPw23DxeGtEWhEaMSqiLuRFpHCyKYoEMWOWhV1L9oqekr0oRhiTHRMVcyT2DGxM2PPxNHiJsXtjHsX7x+/LP5OgnWCJKEtUS1xfGJt4vukgKSVSd3Jo5NnJV9I0U8RpDSnklITU7elDowLHLdmXM941/El469PsJowbcK5ifoTcycemaQ2iTNpfxohLSltZ9oXThSnhjOQzk6vTu/nsrhruS94frzVvF6+N38l/2mGd8bKjGeZ3pmrMnuzfLPKs/oELEGl4FV2SPbG7Pc5UTnbcwZzk3Ib8pTy0vJahJrCHGH7ZKPJ0yZ3iexEJaLuKZ5T1kzpF4eJt+Uj+RPymwu04I98h8Ra8ovkQaFPYVXhh6mJU/dP05gmnNYx3Xb64ulPi4KKfpuBz+DOaJtpMnPezAezmLM2z0Zmp89um2M2Z8GcnrnBc3fMo8zLmfd7sWPxyuK385Pmty4wXDB3waNfgn+pK1EtEZfcWOi1cOMifJFgUedi58XrFn8r5ZWeL3MsKy/7soS75PyvY36t+HVwacbSzmVuyzYsJy4XLr++wnfFjpUaK4tWPloVsapxNX116eq3ayatOVfuUr5xLWWtZG13RXhF8zrzdcvXfanMqrxW5V/VUG1Qvbj6/Xre+ssb/DbUbzTcWLbx0ybBppubgzc31ljWlG8hbinc8mRr4tYzvzF+q92mv61s29ftwu3dO2J3tNe619buNNi5rA6tk9T17hq/69LugN3N9fb1mxt0Gsr2gD2SPc/3pu29vi9sX9t+xv76AxYHqg/SDpY2Io3TG/ubspq6m1Oau1pCW9pavVoPHnI4tP2wyeGqI9pHlh2lHF1wdPBY0bGB46LjfScyTzxqm9R252TyyavtMe2dp8JOnT0ddPrkGeaZY2e9zx4+53mu5TzjfNMFtwuNHa4dB393/f1gp1tn40X3i82XPC61do3tOnrZ9/KJKwFXTl9lX71wLfJa1/WE6zdvjL/RfZN389mt3Fuvbhfe/nxn7l3C3dJ76vfK7xvcr/nD5o+GbrfuIw8CHnQ8jHt45xH30YvH+Y+/9Cx4Qn1S/tT4ae0zp2eHe4N6Lz0f97znhejF576SPzX+rH5p/fLAX35/dfQn9/e8Er8afL3kjd6b7W9d3rYNRA/cf5f37vP70g96H3Z8ZHw88ynp09PPU7+QvlR8tfna+i3s293BvMFBEUfMkf0KYLCiGRkAvN4OADUFABo8n1HGyc9/soLIz6wyBP4Tlp8RZcUNgHr4/x7TB/9ubgCwZys8fkF9tfEARFMBiPcAqLPzcB06q8nOldJChOeATTFf0/PSwb8p8jPnD3H/3AKpqgv4uf0XTIh8SnYVtqgAAAA4ZVhJZk1NACoAAAAIAAGHaQAEAAAAAQAAABoAAAAAAAKgAgAEAAAAAQAAAJygAwAEAAAAAQAAACQAAAAAbKjuNwAAC55JREFUeAHtW01sG8cVniEpwfKfWBj+SwqI7s8haF0xSN2iP6npoGmOYoH+oIdCq0MBu05gKkXPoq51Y1EHF24vXqE9FD3U1LkHUY6BBm4AU02KtEbqUCkSG/5pSEeJLJHc6fdWO+vZ4e6Sktc0VOwCq3l/896bN2/fzOxSnIVc1xZEOplszgjOsgnGJ48+P1AJEY9ZcQS6RoAHSbz1enPKEqzAOUu7MkKY7fbA5LMneN2lxUAcgU1EoCPhlq60xpllFTnnGT89QrA646I4+vzgrB8/psURCIuAm3CoaDnB2BSEc2EdFF4lmUhNfulbvKrQYjCOQGgE+Dt/FZlmsznFODdCJYOY8TIbFJmY7hMBvnRlvSAsPp3gbK8PvycSLbPoX2q1UrNb3d/NFER6MPHpGZYU86fP7o6rZk+R335C9pJ69tTaV775EvvVnn2JlyIYQllwXh5MJhef+QavhenbqK7t0Q9uWC8sv9Mcb7fYsMUTJ1759VAlrN//O+/QoUMGxjiijHPu1q1bNQXvCwg/iqoh+ODBVV6vsLuHM7Ii/VRm5U8v/mRgaN9TyW/3qiBMjiofTrlUreowZFctIUQGy3eG2laTNZYut/atrojPSj1bTbjDhw+XoHNU6kFLk2Qq+LYBMdEVHNqOS4cxrhMYS0XifWoTiGlbtdVqtfbcuXNnRaVtFk7JDmaV1w0mfvTH1xqVzx9NVY//cJAP7eLqBErRnlvnlUqOOuBAkrc7gthuW9f/9Wa7fu+myNq0CP5gUrLaJC1GoDZWEXEE3IQjvXbSZUXu3281ajfebg9/7svJ6gs/HmwPDvHnorBrWeyD//xTvPf+9VYkFTQKn2Id/Y2AJ+HINCXdya9+nGdJawFJl73x9io79t3BK8+9mDqSSLKnt+IeltbGvQ/FtevXms9inxZpsmH5sZceVLdhzbcRyWu32w0sBfaSrsqkce3YsWMMtAzuHHRULcuqoZ332zM58m7VV/XC1rijh6HaYivhrwMy9gV52laMQS7tkGoPHjyYr+Ny8K7NZv0nhTImUjnGaa8Eqv/g0XbkfSmjtnoMiCd1qHIE67ZWV1f/znUhiZ/8WsNknFEQ7YuC+L2f7vz9F7PsO4z1utSKpfv3+J//8UbrB62mdVTqCms3u4fDPgOrdfgF3xcRlJyUoqANDQ1NAS9Imk9bQoCm1QRAACkpF6Qs6UXSFVKp1EXQspIuW/AnYNeUuGyhZwp6ihJX2jr6TAI3wD8u6aB17OFCdNjd0MdEAk+q/hNDjxfG+BnEgsbk+u/Yu+y3h8MrtBTkP7KNOH8Qg+/fvn27rNLgXwZjeE+loe/TCZXggRODRRVH5/Rf/rD6ym9+sXppIJU6wgSfEIxNQ4aeEPcGbZZ4JPP6pfVS9fLaZK/Jptp7nLAT4LBkI/MFRy7QFcRkOJlMXoKAO1mqMPgzFHiVBrwIelGlKXAavIu43Sqq8FwQOhZCdNhy4BvkPz1cbkcfADIXQfb130ecOQk8r/IQg5yKEwz7eZWGJF6+e/fuhx1LqhS68MZQDVVuTq1yNo+z4uyZ+/zC1eGilPVrz7+6YsDqRe7HfII0PLUlmNcDvISA0DKYBk+d7CzJ37x5sxDgMh1UAlg2mfQZuIu4aYmhp36KYPWC7QboNdCkbernezn+5zRmoP9IqCKSJMh/UpPXdHVFUdFMJNmYFIT/BHtsgJZXYwO4RPLBFQ5Mi/EyCXVcnE2dPHbf6KA7hPO/XMmi0s0E8aOkIxk43RggVVn3Aj4teXI5dZ72M64QAEcuSzKQzwKfUPmAx7tUCZrsI44PJwA3tP45iSPonkkhOtmD7TTZpuUNpCUpr7eOH+MqHf3nqK/0n5Y3lQ/4TBf/yYcG7mnnnoOOuqbDg9LySX0kEePK7N+/PytxsgfacYlTC3k7l0IT7ndX99pCakcJC27N/OzYimtE0u3WspeFtIfWEyIaCR4+2J7UBAjhaTc01hImqqjSgJvA1Umng0VelVFhBDKPPjWioa2gKRDsd0HWEy/gc449W5yWK9Ln15dojh9pyYfsMvobEqfW2UvNqzSfcbts0oG9XgZ6is5t+B2w3A4OgIQyVRr2sYbEfeLlHsICl1TZGam5iKXRk63Eg8F0kreoip1wZQGcf/WTIhpPYFV+EIyKOJdsJ4unXhuqBck8Kh3BpSfPVQP8IyxzUy7BAYiuyoGc0WUIh1wDk1TTeDrusqFTj6PpMh2A9MGnRR9Zksg4YnYDmSD/h8FzRWncLqIBOJUXKNE1clcUL4FNJJm6Wrhjg+28qgD2yxLvnnCMVyDsKpMdN1qeo6X1wt/2moTT91DBPjnDsfHr+aKETrLCy/35fppT/UJgCPfQVL6EIRc0YVUpE3Fbg76AmHss0R4y66H4IGEy+unSp7sviaog9pO0Eow6Alnao6Ja1oGPyU7OQ2lKPHRJtYWsREUK+7VYWt0sH0isGiGT4+kuGFumVyCnz+3O9fFjfc3jRO9IpnfRSCQjtRdW4R7RW1Ptj7nP+yynZVWmhwqninfC9PTgRXHuwpt7KpwJN/k6JSUFm038kvjlc7tNSeljW1NtYSJoD2WotMcMqxWBTGUC7I0E0HXyPA4MeZ3YLxyHHBP7wxnF3jhgGqN74QBSchEA3SucKh0Ai6SVp5MpVtJMgMgGWbD59fauDKqaGSq3BSYSv6p1y2g4fshseWTQZ4yWAV0OS8Ul0BfkrZ7AdNnN4Ehwj330pQnyXAcOHKDXCRkP0UEweWWN7us//L4ofXfanNYvEtTZ+80rymiJV5fTZf0A0rXCUeU6+fWGotMHFOK4aCdqnGOh9L3wnolx4+fndukB85XeChGTWcdg3a6AxxFscqiGm05zJu1XQFsGb8QRTAO+dPDgweLa2triwMBABhvhKfBo0m0R6PX9LOb032xTQYdx2Qk2cpQQsFECTP7nwStIvt767JtIZAH+F3T/ZV/yH/uqqsSjbvEQeN7JQX9a2sB4ShKWbdeEk4JhLRSjuglPKZXy+I661LR25SZLj/0fb0zYpGRxL/hlEIKgL6IxCaZTmfN1gFC6solEooylYQPT/kJHUSNtGaWkR/WkhBqVSqA/R7fEu7U4HRp4KK5JOfTN4A70H3KlrZxCpf5urfMQ04vrYV0WcS/rtEiWVFLaWudf0JXTK5WmtbMfyUbvwGqwP9vhg0agACEQExo5CJ3FHqnjKQ0S7oVOCUNVJ0SWHlzfh5f6UJUj/7vosNVDhvaoRRt5jH+QbKaPevfdm8qLpMKRwtWPrd0D+zaWIcIF3qvhYGAQ3K8LyVHAElWHPTotj0i7gInmXlRpsC+rolIUQRxzGQ6AiaKKSC9CKyqP3sCjOhLPvqC3KmHZdpOhhIHtHPQU0N9dXp0EKmP5o2+4ReCuz6RT6qdW+k86gNLy76kuQf5TX4dHYLfLgmxFFYLvnh9kSp7POzmyU5Z8tX2YISpVg7GHQ/6EXyPPJJf27tv46LyxjPansgV5RZ9XaE9GExwkI+lI0hzBNLG9yMt+UbR0IEHipPXk3oxuqeNJ+E9+kn11mUey0QvxtN8YIqtwD5XjgJBkxuQ5b1V5yO8P5OxbuiYbefMok/2oo4kiwaPQ8SjjcCqtqsK3upFA9AkneOn02V09TbTqYQxvzwhgdaBDi7s1oFGg0paCRhPZoWHDgGisWzsDjQU5EdO3dQQM1Xsspx3v3lR+tBUO1a0Prz9U/2P4CUcA1S2LJHMPUnDHDHMp0oRLWIlQY2GOxLztGYHNflrrbUnlg/hJuSeLO6LTbvF3T5Ue30+LOgzGhG0ZgZ4Sjn5ufuFqOsfaCfpFq+9LybVPN/7ReVtGIXa6bxHoKeGkN/Rd9bdX01lm8Qm8QlyWdGpXV6x3VTyG4wj4RWBTCScV0A8uH6wNZ/E5gX4Hb3+mWfkvuyX5cRtHICgCW0o4Ukb/ME3/ubW2PpxB4s0FGYjpcQTUCPwPioM0XXAoKPAAAAAASUVORK5CYII="/>
</defs>
</svg>
