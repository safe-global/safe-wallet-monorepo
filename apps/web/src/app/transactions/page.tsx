export { default } from '@/legacy-pages/transactions'
