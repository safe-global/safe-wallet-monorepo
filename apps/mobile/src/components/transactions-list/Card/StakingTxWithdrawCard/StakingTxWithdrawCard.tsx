import { SafeListItem } from '@/src/components/SafeListItem'
import { TokenAmount } from '@/src/components/TokenAmount'
import { NativeStakingWithdrawTransactionInfo } from '@safe-global/store/gateway/AUTO_GENERATED/transactions'
import { Logo } from '@/src/components/Logo'

export const StakingTxWithdrawCard = ({ info }: { info: NativeStakingWithdrawTransactionInfo }) => {
  return (
    <SafeListItem
      label={`Claim`}
      icon="transaction-stake"
      type={'Stake'}
      rightNode={
        <TokenAmount
          testID="token-amount"
          value={info.value}
          tokenSymbol={info.tokenInfo.symbol}
          decimals={info.tokenInfo.decimals}
        />
      }
      leftNode={<Logo logoUri={info.tokenInfo.logoUri} accessibilityLabel={info.tokenInfo.symbol} />}
    />
  )
}
