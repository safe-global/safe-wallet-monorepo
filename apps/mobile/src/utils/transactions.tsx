import { GroupedTxs } from '@/src/features/TxHistory/utils'
import {
  isTransferTxInfo,
  isAddSignerTxInfo,
  isMultiSendTxInfo,
  isCustomTxInfo,
  isERC721Transfer,
  isSwapOrderTxInfo,
  isRemoveSignerTxInfo,
} from '@/src/utils/transaction-guards'
import { Transaction } from '@safe-global/store/gateway/AUTO_GENERATED/transactions'
import { ETxType } from '../types/txType'

export const getTransactionType = ({ txInfo }: { txInfo: Transaction['txInfo'] }) => {
  if (isTransferTxInfo(txInfo)) {
    return ETxType.TOKEN_TRANSFER
  }

  if (isTransferTxInfo(txInfo) && isERC721Transfer(txInfo)) {
    return ETxType.NFT_TRANSFER
  }

  if (isAddSignerTxInfo(txInfo)) {
    return ETxType.ADD_SIGNER
  }

  if (isRemoveSignerTxInfo(txInfo)) {
    return ETxType.REMOVE_SIGNER
  }

  if (isMultiSendTxInfo(txInfo) || isCustomTxInfo(txInfo)) {
    return ETxType.CONTRACT_INTERACTION
  }

  if (isSwapOrderTxInfo(txInfo)) {
    return ETxType.SWAP_ORDER
  }

  return null
}

export const groupBulkTxs = <T extends { type: string; transaction?: Transaction }>(
  list: GroupedTxs<T>,
): GroupedTxs<T> => {
  const grouped = list
    .reduce<GroupedTxs<T>>((resultItems, item) => {
      if (Array.isArray(item) || item.type !== 'TRANSACTION') {
        return resultItems.concat([item])
      }
      const currentTxHash = item.transaction?.txHash

      const prevItem = resultItems[resultItems.length - 1]
      if (!Array.isArray(prevItem)) {
        return resultItems.concat([[item]])
      }
      const prevTxHash = prevItem[0]?.transaction?.txHash

      if (currentTxHash && currentTxHash === prevTxHash) {
        prevItem.push(item)
        return resultItems
      }

      return resultItems.concat([[item]])
    }, [])
    .map((item) => (Array.isArray(item) && item.length === 1 ? item[0] : item))

  return grouped
}
