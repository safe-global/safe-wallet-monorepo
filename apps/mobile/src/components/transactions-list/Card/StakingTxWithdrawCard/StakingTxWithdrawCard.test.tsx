import React from 'react'
import { render } from '@/src/tests/test-utils'
import { StakingTxWithdrawCard } from './StakingTxWithdrawCard'
import { NativeStakingWithdrawTransactionInfo } from '@safe-global/store/gateway/AUTO_GENERATED/transactions'

describe('StakingTxWithdrawCard', () => {
  const mockInfo = {
    value: '1000000000000000000',
    tokenInfo: {
      symbol: 'ETH',
      decimals: 18,
      logoUri: 'https://example.com/eth-logo.png',
      name: 'Ethereum',
      address: '0x0000000000000000000000000000000000000000',
    },
  } as NativeStakingWithdrawTransactionInfo

  it('renders correctly', () => {
    const { toJSON } = render(<StakingTxWithdrawCard info={mockInfo} />)
    expect(toJSON()).toMatchSnapshot()
  })

  it('renders correctly with given info', () => {
    const screen = render(<StakingTxWithdrawCard info={mockInfo} />)

    // Check that important props are passed correctly
    expect(screen.getByText('Claim')).toBeTruthy()
    expect(screen.getByTestId('token-amount')).toHaveTextContent('1 ETH')
    expect(screen.getByTestId('logo-image')).toBeTruthy()
  })
})
