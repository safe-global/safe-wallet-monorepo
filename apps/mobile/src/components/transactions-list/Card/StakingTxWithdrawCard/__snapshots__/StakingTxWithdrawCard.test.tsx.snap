// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StakingTxWithdrawCard matches snapshot 1`] = `
<View>
  <View
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "flex-start",
        "backgroundColor": "#FFFFFF",
        "borderBottomLeftRadius": 7,
        "borderBottomRightRadius": 7,
        "borderTopLeftRadius": 7,
        "borderTopRightRadius": 7,
        "flexDirection": "column",
        "flexWrap": "wrap",
        "gap": 12,
        "justifyContent": "flex-start",
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 12,
            "maxWidth": "55%",
          }
        }
      >
        <View
          style={
            {
              "width": 32,
            }
          }
        >
          <View
            style={
              {
                "position": "absolute",
                "right": -10,
                "top": -10,
                "zIndex": 1,
              }
            }
          />
          <View
            style={
              {
                "alignItems": "center",
                "borderBottomLeftRadius": 100000,
                "borderBottomRightRadius": 100000,
                "borderTopLeftRadius": 100000,
                "borderTopRightRadius": 100000,
                "flexDirection": "column",
                "height": 32,
                "justifyContent": "center",
                "maxHeight": 32,
                "maxWidth": 32,
                "minHeight": 32,
                "minWidth": 32,
                "overflow": "hidden",
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 0,
                "position": "relative",
                "width": 32,
              }
            }
          >
            <View
              style={
                {
                  "bottom": 0,
                  "flexDirection": "column",
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "zIndex": 1,
                }
              }
            >
              <Image
                accessibilityLabel="ETH"
                fullscreen={true}
                onError={[Function]}
                onLoad={[Function]}
                source={
                  {
                    "uri": "https://safe-transaction-assets.safe.global/chains/1/chain_logo.png",
                  }
                }
                style={
                  {
                    "backgroundColor": "#121312",
                    "height": 32,
                    "width": 32,
                  }
                }
                testID="logo-image"
              />
            </View>
            <View
              style={
                {
                  "backgroundColor": "#EEEFF0",
                  "bottom": 0,
                  "flexDirection": "column",
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "zIndex": 0,
                }
              }
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#EEEFF0",
                    "borderBottomLeftRadius": 0,
                    "borderBottomRightRadius": 0,
                    "borderTopLeftRadius": 0,
                    "borderTopRightRadius": 0,
                    "display": "flex",
                    "height": 32,
                    "justifyContent": "center",
                    "width": 32,
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    [
                      {
                        "color": "#A1A3A7",
                        "fontSize": 16,
                      },
                      undefined,
                      {
                        "fontFamily": "SafeIcons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      {},
                    ]
                  }
                  testID="logo-fallback-icon"
                >
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View>
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "gap": 4,
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "#A1A3A7",
                    "fontSize": 10,
                  },
                  undefined,
                  {
                    "fontFamily": "SafeIcons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
              testID="safe-list-transaction-stake-icon"
            >
              
            </Text>
            <Text
              style={
                {
                  "color": "#A1A3A7",
                  "fontFamily": "DM Sans",
                  "fontSize": 12,
                  "lineHeight": 20,
                }
              }
              suppressHighlighting={true}
            >
              Stake
            </Text>
          </View>
          <Text
            style={
              {
                "color": "#121312",
                "fontFamily": "DMSans-SemiBold",
                "fontSize": 14,
                "letterSpacing": -0.01,
                "lineHeight": 20,
              }
            }
            suppressHighlighting={true}
          >
            Claim
          </Text>
        </View>
      </View>
      <Text
        style={
          {
            "color": "#121312",
            "fontFamily": "DMSans-Bold",
          }
        }
        suppressHighlighting={true}
        testID="token-amount"
      >
        32
         
        ETH
      </Text>
    </View>
  </View>
</View>
`;
